pkg/util/errors/errors.go

```diff
@@ -29,10 +29,16 @@ type Aggregate interface {
 // is itself an implementation of the error interface.  If the slice is empty,
 // this returns nil.
 func NewAggregate(errlist []error) Aggregate {
-	if len(errlist) == 0 {
+	errs := []error{}
+	for _, err := range errlist {
+		if err != nil {
```

Comment:
The main reason behind this change is that I don't want to precheck if an error is nil before passing it to the list that will be aggregated. Unconditionally passing errors is very helpful and reduces a lot the code I need to write. If we continue w/o this change or by returning a nil string which will still be an error then there is no much gain from this change (with respect to unconditional errors aggregating).